d3b_utils/s3_contents.py

```diff
@@ -0,0 +1,182 @@
+import csv
+import os
+import re
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from itertools import chain
+
+import boto3
+
+
+def fetch_obj_list_info(pathlist, profile="saml", all_versions=False):
```

Comment:
because the defaults for profile + all_versions, why not use **kwarqs instead?
```suggestion
def fetch_obj_list_info(pathlist, **kwarqs):
```

If we do that, we will _need_ to handle what happens when a user specifies `output_filename`...

Reply:
I kinda don't like kwargs because I don't like having to look somewhere else to find out what a function takes as input.

Reply:
good points.